File: lib/src/native/video_recorder/record_error.dart
class RecordError implements Exception {

  RecordError({
    required this.code,
    this.message,
    this.details,
  });

  factory RecordError.fromJson(Map<String, dynamic> json) {
    return RecordError(
      code: json['code'],
      message: json['message'],
      details: json['details'],
    );
  }
  final String code;
  final String? message;
  final Map<String, dynamic>? details;
}

File: android/src/main/java/com/cloudwebrtc/webrtc/videoRecorder/RecordError.kt
package com.cloudwebrtc.webrtc.videoRecorder

data class RecordError(
    val code: String,
    val message: String?,
    val details: Map<String, String>? = null

) {
    fun toMap(): Map<String, Any?> {
        return mapOf(
            "code" to code,
            "message" to message,
            "details" to details
        )
    }
}
